examples/app/headless_renderer.rs

```diff
@@ -0,0 +1,444 @@
+                    size,
+                    dimension: TextureDimension::D2,
+                    format: TextureFormat::Rgba8UnormSrgb,
+                    mip_level_count: 1,
+                    sample_count: 1,
+                    usage: TextureUsages::COPY_DST | TextureUsages::TEXTURE_BINDING,
+                    view_formats: &[],
+                },
+                ..Default::default()
+            };
+            cpu_image.resize(size);
+            let cpu_image_handle = images.add(cpu_image);
+
+            commands.spawn(ImageCopier::new(
+                render_target_image_handle.clone(),
+                cpu_image_handle.clone(),
+                size,
+                render_device,
+            ));
+
+            commands.spawn(ImageToSave(cpu_image_handle));
+
+            scene_controller.state = SceneState::Render(pre_roll_frames);
+            scene_controller.name = scene_name;
+            RenderTarget::Image(render_target_image_handle)
+        }
+
+        fn update(
+            images_to_save: Query<&ImageToSave>,
+            mut images: ResMut<Assets<Image>>,
+            mut scene_controller: ResMut<SceneController>,
+            mut app_exit_writer: EventWriter<AppExit>,
+        ) {
+            if let SceneState::Render(n) = scene_controller.state {
+                if n < 1 {
+                    use rand::Rng;
+                    let mut rng = rand::thread_rng();
+                    for image in images_to_save.iter() {
+                        let img_bytes = images.get_mut(image.id()).unwrap();
+
+                        let img = match img_bytes.clone().try_into_dynamic() {
+                            Ok(img) => img.to_rgba8(),
+                            Err(e) => panic!("Failed to create image buffer {e:?}"),
+                        };
+
+                        let images_dir =
+                            PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("test_images");
+                        println!("Saving image to: {images_dir:?}");
+                        std::fs::create_dir_all(&images_dir).unwrap();
+
+                        let number = rng.gen::<u128>();
+                        let image_path = images_dir.join(format!("{number:032X}.png"));
+                        if let Err(e) = img.save(image_path) {
+                            panic!("Failed to save image: {}", e);
+                        };
+                    }
+                    if scene_controller.single_image {
+                        app_exit_writer.send(AppExit);
+                    }
+                } else {
+                    scene_controller.state = SceneState::Render(n - 1);
+                }
+            }
+        }
+    }
+}
+
+struct AppConfig {
+    width: u32,
+    height: u32,
+    single_image: bool,
+}
+
+use bevy::{
+    app::ScheduleRunnerPlugin, core_pipeline::tonemapping::Tonemapping, prelude::*,
+    render::renderer::RenderDevice,
+};
+
+fn main() {
+    let mut app = App::new();
```

Comment:
It's pretty minor, but I'd prefer if this used chaining instead of having `app.` on each line. It's mainly to be more consistent with other examples.